public/react/services/reports/useSaveReports.tsx

```diff
@@ -40,7 +40,16 @@ const useSaveReport = () => {
   return { save, isPending };
 };
 
-export const useCreateReport = () => {
+export const useCreateReport = (
+  {
+    onCreate,
+  }: {
+    /**a function that gets run when the report is saved successfully */
+    onCreate: (newReportId: string) => Promise<void>;
+  } = {
+    onCreate: (_: string) => Promise.resolve(),
```

Comment:
There shouldn't be a need to default this. If its not needed it can just be optional